specutils/io/default_loaders/sdss.py

```diff
@@ -113,38 +159,96 @@ def spSpec_loader(file_obj, **kwargs):
+    if 'Ang' in bunit and 'strom' not in bunit:
+        bunit = bunit.replace('Ang', 'Angstrom')
+    flux_unit = Unit(bunit)
+    flux = hdulist[0].data[0, :] * flux_unit
+
+    uncertainty = StdDevUncertainty(hdulist[0].data[2, :] * flux_unit)
+
+    # dispersion along NAXIS1 from the WCS
+    dispersion = wcs.pixel_to_world(np.arange(flux.shape[0]))
+    # convert out of logspace (default for spSpec/spPlate spectra)?
+    if header.get('DC-Flag', 1) == 1:
+        dispersion = 10**dispersion
+    dispersion_unit = Unit('Angstrom')
 
-    data = hdulist[0].data[0, :]
-    unit = Unit('1e-17 erg / (Angstrom cm2 s)')
+    mask = hdulist[0].data[3, :] != 0
 
-    uncertainty = StdDevUncertainty(hdulist[0].data[2, :] * unit)
+    if not isinstance(file_obj, fits.hdu.hdulist.HDUList):
+        hdulist.close()
 
-    # dispersion from the WCS but convert out of logspace
-    # dispersion = 10**wcs.all_pix2world(np.arange(data.shape[0]), 0)[0]
-    dispersion = 10**wcs.all_pix2world(np.vstack((np.arange(data.shape[0]),
-                                                  np.zeros((data.shape[0],)))).T,
-                                       0)[:, 0]
-    # dispersion = 10**hdulist[1].data['loglam']
+    return Spectrum1D(flux=flux, spectral_axis=dispersion * dispersion_unit,
+                      uncertainty=uncertainty, meta=meta, mask=mask)
+
+
+@data_loader(label="SDSS spPlate", identifier=spPlate_identify, extensions=['fits'])
+def spPlate_loader(file_obj, limit=None, **kwargs):
+    """
+    Loader for SDSS spPlate files, reading flux spectra from all fibres into single array.
+
+    Parameters
+    ----------
+    file_obj: str, file-like, or HDUList
+           FITS file name, object (provided from name by Astropy I/O Registry),
+           or HDUList (as resulting from astropy.io.fits.open()).
+
+    limit : :class:`int`, optional
+        If set, only return the first `limit` spectra in `flux` array.
+
+    Returns
+    -------
+    Spectrum1D
+        The spectra represented by the wavelength solution from the header WCS
+        and the data array of the primary HDU (typically 640 along dimension 1).
+    """
+    if isinstance(file_obj, fits.hdu.hdulist.HDUList):
+        hdulist = file_obj
+    else:
+        hdulist = fits.open(file_obj, **kwargs)
+
+    header = hdulist[0].header
+    meta = {'header': header}
+    wcs = WCS(header).dropaxis(1)
+    if limit is None:
+        limit = header['NAXIS2']
+
+    bunit = header.get('BUNIT', '1e-17 erg / (Angstrom cm2 s)')
+    if 'Ang' in bunit and 'strom' not in bunit:
+        bunit = bunit.replace('Ang', 'Angstrom')
+    flux_unit = Unit(bunit)
+    flux = hdulist[0].data[0:limit, :] * flux_unit
+    uncertainty = InverseVariance(hdulist[1].data[0:limit, :] / flux_unit**2)
+
+    # dispersion along NAXIS1 from the WCS
+    wcs = WCS(header).dropaxis(1)
+    dispersion = wcs.pixel_to_world(np.arange(flux.shape[-1]))
+    # convert out of logspace (default for spSpec/spPlate spectra)?
```

Comment:
Yes log(lambda) is the default for spPlate, spSpec and spec files.  This is just a SDSS "thing".

Reply:
The only change here is really the check for `DC-Flag`, which I took from the documentation is the card indicating the logical (although I could not find ultimate confirmation). Anyway the default is left at "1" for "loglinear", but should the loader encounter a spectrum with `DC-Flag = 0`, it would assume linear dispersion.